api-admin/src/main/java/bio/terra/pearl/api/admin/service/auth/BaseEnforcePermissionAspect.java

```diff
@@ -1,14 +1,17 @@
 package bio.terra.pearl.api.admin.service.auth;
 
+import bio.terra.pearl.api.admin.service.auth.context.PortalAuthContext;
 import java.lang.annotation.Annotation;
 import org.apache.commons.lang3.NotImplementedException;
+import org.aspectj.lang.JoinPoint;
 import org.aspectj.lang.ProceedingJoinPoint;
 
 /** base class for annotations enforcing permissions around methods. */
 public abstract class BaseEnforcePermissionAspect<
     T extends PortalAuthContext, A extends Annotation> {
+  private static final String ANNOTATION_NAME = "EnforcePermission";
   private static final String AUTH_CONTEXT_USAGE_ERROR =
-      "EnforcePortalPermission annotation must be used on a method whose first argument is a %s";
+      "%s annotation must be used on a method whose first argument is a %s";
```

Comment:
Would it be possible to have `@SandboxOnly` and also `@EnforcePortalPermission` at the same time? Since we're using argument types we know (`<something>AuthContext`), maybe these could search for that type instead of only being in the first argument for even more cool auto-magic composability.

Reply:
yes!  these are designed to be composable.  The idea is that there will be only one authContext argument (the first arg), and that can be either a PortalAuthContext or an EnvironmentAwareAuthContext, or *both*, since EnvironmentAwareAuthContext is just an interface